server/src/main/java/org/elasticsearch/action/admin/indices/sampling/SamplingConfiguration.java

```diff
@@ -0,0 +1,211 @@
+        PARSER.declareIntOrNull(optionalConstructorArg(), DEFAULT_MAX_SAMPLES, new ParseField(MAX_SAMPLES_FIELD_NAME));
+        PARSER.declareField(
+            optionalConstructorArg(),
+            (p, c) -> ByteSizeValue.parseBytesSizeValue(p.text(), MAX_SIZE_FIELD_NAME),
+            new org.elasticsearch.xcontent.ParseField(MAX_SIZE_FIELD_NAME),
+            org.elasticsearch.xcontent.ObjectParser.ValueType.STRING
+        );
+        PARSER.declareField(
+            optionalConstructorArg(),
+            (p, c) -> TimeValue.parseTimeValue(p.text(), TIME_TO_LIVE_FIELD_NAME),
+            new org.elasticsearch.xcontent.ParseField(TIME_TO_LIVE_FIELD_NAME),
+            org.elasticsearch.xcontent.ObjectParser.ValueType.STRING
+        );
+        PARSER.declareStringOrNull(optionalConstructorArg(), new org.elasticsearch.xcontent.ParseField(CONDITION_FIELD_NAME));
+    }
+
+    /*
+     * Constructor with validation and defaulting for optional fields.
+     */
+    public SamplingConfiguration(double rate, Integer maxSamples, ByteSizeValue maxSize, TimeValue timeToLive, String condition) {
+        validateInputs(rate, maxSamples, maxSize, timeToLive, condition);
+
+        // Set defaults
+        maxSamples = maxSamples == null ? DEFAULT_MAX_SAMPLES : maxSamples;
+        maxSize = maxSize == null ? ByteSizeValue.ofGb(DEFAULT_MAX_SIZE_GIGABYTES) : maxSize;
+        timeToLive = timeToLive == null ? TimeValue.timeValueDays(DEFAULT_TIME_TO_LIVE_DAYS) : timeToLive;
+
+        // Initialize record fields
+        this.rate = rate;
+        this.maxSamples = maxSamples;
+        this.maxSize = maxSize;
+        this.timeToLive = timeToLive;
+        this.condition = condition;
+    }
+
+    // StreamInput constructor
+    public SamplingConfiguration(StreamInput in) throws IOException {
+        this(
+            in.readDouble(),
+            in.readOptionalInt(),
+            in.readOptionalWriteable(ByteSizeValue::readFrom),
+            in.readOptionalTimeValue(),
+            in.readOptionalString()
+        );
+    }
+
+    // Write to StreamOutput
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeDouble(this.rate);
+        out.writeOptionalInt(this.maxSamples);
+        out.writeOptionalWriteable(this.maxSize);
+        out.writeOptionalTimeValue(this.timeToLive);
+        out.writeOptionalString(this.condition);
+    }
+
+    // Serialize to XContent (JSON)
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(RATE_FIELD_NAME, rate);
+        if (maxSamples != null) {
+            builder.field(MAX_SAMPLES_FIELD_NAME, maxSamples);
+        }
+        if (maxSize != null) {
+            builder.field(MAX_SIZE_FIELD_NAME, maxSize.toString());
+        }
+        if (timeToLive != null) {
+            builder.field(TIME_TO_LIVE_FIELD_NAME, timeToLive.toString());
+        }
```

Comment:
There is a trick I haven't told you about for TimeValue and ByteSizeValue fields -- use builder.humanReadableField(). That way when we use this for the get sample config rest action, we can pass a human-readable flag, and it will display the value in a way that's easier to read.